src/hooks/useJettonLogo.tsx

```diff
@@ -0,0 +1,78 @@
+import { useEffect } from "react";
+import coinLogoHover from "assets/icons/coin-logo-hover.svg";
+import coinLogo from "assets/icons/coin-logo.svg";
+import { atom, useRecoilState } from "recoil";
+import jet from "assets/icons/logo.svg";
+
+const jettonLogoState = atom({
+  key: "jettonLogo",
+  default: {
+    iconHover: false,
+    logoUrl: "",
+    imageValidationError: false,
+    image: "",
+    tempUrl: "",
+  },
+});
+
+export const useJettonLogo = () => {
+  const [jettonLogo, setJettonLogo] = useRecoilState(jettonLogoState);
+
+  const setLogoUrl = (val: string) =>
+    setJettonLogo((prev) => {
+      return {
+        ...prev,
+        logoUrl: val,
+      };
+    });
+
+  const setTempUrl = (val: string) =>
+    setJettonLogo((prev) => {
+      return {
+        ...prev,
+        tempUrl: val,
+      };
+    });
```

Comment:
I think a better approach would be to save all edits in a single variable (jettonURL) and if the user closes the edit popup and hasn't saved we can show an "Are you sure / your changes will not be saved" popup